python/pyspark/pandas/frame.py

```diff
@@ -10232,7 +10248,15 @@ def all(self, axis: Axis = 0, bool_only: Optional[bool] = None) -> "Series":
         applied = []
         for label in column_labels:
             scol = self._internal.spark_column_for(label)
-            all_col = F.min(F.coalesce(scol.cast("boolean"), SF.lit(True)))
+
+            if isinstance(self._internal.spark_type_for(label), NumericType) or skipna:
```

Comment:
Similar to [Series.all](https://github.com/apache/spark/blob/master/python/pyspark/pandas/base.py#L1041-L1048)